### ISSUE TITLE
[Toolkit bug]: Unloading a stage does not free vram + issue when loading captures

### ISSUE THREAD

**Issue by xoxor4d** [NVIDIAGameWorks/rtx-remix#484]
### Describe the bug

From the latest changelog: 
`Added the option to unload the stage to reclaim resources without closing the app. This function is useful if you need to free up memory to do certain actions (for example, boot the game while running RTX Remix, or run RTX Remix’s AI Texture Tools). This option can be found in the top left menu.`

Unloading a stage does not free my gpu's vram so there is no chance to load up the game or think about any texture upscaling without closing the toolkit first.

Loading a capture with the stage unloaded results in the project failing to load / throwing an error message.

https://github.com/NVIDIAGameWorks/rtx-remix/assets/45299104/56308722-ff32-4095-ae24-5518eaae7f42

### How do you reproduce the bug?

1. Unload a stage and watch vram usage
2. Load a capture after stage unloading

### What is the expected behavior?

The toolkit should free gpu vram.
Loading a capture after unloading a stage should reload the stage.

Maybe throw the user into the homescreen when unloading a stage so that he has to re-load the project from there

### Version

2024.3.0

**Comment by sfitzpatrickchapman** [comment]
REMIX-3558 for tracking

**Comment by NV-LL** [comment]
REMIX-4885 for Runtime side of the issue, as well

**Comment by CAnguianoNV** [comment]
Hi @xoxor4d — thank you for reporting this, and for your patience while we investigated.

We've confirmed that unloading a stage does not currently release the associated VRAM. Fully addressing this will require a broader change than initially anticipated, so we'd like to better understand the real-world impact and properly evaluate the work.

Could you share the following information?

- **GPU details**
  - GPU model
  - Total VRAM
  - RTX Remix Toolkit version

- **Approximate VRAM usage**
  - Before starting the Toolkit
  - After starting the Toolkit, but before opening a project
  - After loading a project

- **Workflow impact**
  - What do you typically need to do after unloading the stage—for example, launch the game, use the AI Texture Tools, or load another capture?
  - What happens if you continue without restarting the Toolkit—for example, does the game fail to launch, crash, run poorly, or run out of available VRAM?
  - How many times during a typical session or day do you need to restart the Toolkit?
  - Approximately how long does each restart take, including reopening the project and restoring your working context?
  - In a hypothetical eight-hour work session, approximately how much total time does this cost you?

Approximate measurements are completely fine. This information will help us quantify the workflow impact and determine the best path toward improving it.

Thanks again for helping us investigate.

**Comment by xoxor4d** [comment]
Hey @CAnguianoNV, sorry for the late reply - I was on vacation.

Back then, I was using a RTX 3080 with 10 or 12gb of VRAM. I don't remember the specific version of the toolkit but I was probably running the very latest commit around the day of creating this issue.

> What do you typically need to do after unloading the stage—for example, launch the game, use the AI Texture Tools, or load another capture?

Since I was constraint on VRAM, I wanted a way to quickly release the VRAM used by the Toolkit without completely closing the toolkit so I could use AI Upscaling tools / Photoshop or other software like Substance Painter. 

> What happens if you continue without restarting the Toolkit—for example, does the game fail to launch, crash, run poorly, or run out of available VRAM?

Other software would just not work or be really slow. Running the game next to the toolkit was almost impossible. It was only an option with a very, very tiny loaded capture - and only if that capture was the first to be loaded after opening the toolkit.

> How many times during a typical session or day do you need to restart the Toolkit?

That really depends on what I was working on. I sometimes launched the toolkit 1-3 times for a single material to see what it looked like iirc. 
It was not really about the time it took, it was just a tedious experience having to restart the toolkit each time. 

> Approximately how long does each restart take, including reopening the project and restoring your working context?

I think the toolkit was a little slower to launch back then. I don't remember how long exactly tho. It now takes around 10seconds for a relaunch (after it was opened at least once). Launching a project really depends on the size of the project and the capture itself and can range between 5s to 30s for current projects of mine.

--------------

**Thoughts today:**
If the toolkit does not retain meaningful amounts of data in system memory that increases loading times of captures - and most of the time is spent on uploading stuff to gpu-VRAM, then the unloading stage logic (closing a project) most likely does not have much benefit. 

I have 32gb of VRAM available now and can run the toolkit in parallel with other software so it's no longer a concern for me, but not everybody has that luxury. 

<br>

**Idea:**
Maybe a low VRAM option for the toolkit might be viable option. Eg: Loading / displaying everything in a capture in a lower MIP and only loading the highest MIP for the object that is currently selected? This is obv. not part of this issue. So if you feel like that might be something that could be done and sounds reasonable, let me know and I'll create a separate "Issue" for that

**Comment by CAnguianoNV** [comment]
@xoxor4d  Thank you so much for the thoughtful feedback. This is all very useful, and I’ll make sure to share it with the team.

Really appreciate you taking the time to provide your input.

Cheers!
